components/brave_wallet/browser/json_rpc_service.cc

```diff
@@ -2181,12 +2178,56 @@ void JsonRpcService::EthGetLogs(const std::string& chain_id,
     return;
   }
 
+  const auto load_params = [&](std::string& from_block, std::string& to_block,
+                               std::string& block_hash,
+                               base::Value::List* address,
+                               base::Value::List* topics) {
+    auto* filtering_opts = params.GetIfDict();
+    if (filtering_opts == nullptr) {
+      return;
+    }
+
+    const base::Value* from_block_val = filtering_opts->Find("fromBlock");
+    if (from_block_val != nullptr && from_block_val->is_string()) {
+      from_block = from_block_val->GetString();
+    }
+
+    const base::Value* to_block_val = filtering_opts->Find("toBlock");
+    if (to_block_val != nullptr && to_block_val->is_string()) {
+      to_block = *(to_block_val->GetIfString());
+    }
+
+    const base::Value* block_hash_val = filtering_opts->Find("blockHash");
+    if (block_hash_val != nullptr && block_hash_val->is_string()) {
+      block_hash = *(block_hash_val->GetIfString());
+    }
+
+    const auto* contract_addresses_str = filtering_opts->FindString("address");
+    if (contract_addresses_str != nullptr) {
+      address->Append(*contract_addresses_str);
+    }
+
+    const base::Value::List* contract_addresses =
+        filtering_opts->FindList("address");
+    if (contract_addresses != nullptr) {
+      *address = contract_addresses->Clone();
+    }
+
+    const base::Value::List* topics_list = filtering_opts->FindList("topics");
+    if (topics_list != nullptr) {
+      *topics = topics_list->Clone();
+    }
+  };
+
+  std::string from_block, to_block, block_hash;
+  base::Value::List contract_addresses, topics;
+  load_params(from_block, to_block, block_hash, &contract_addresses, &topics);
   auto internal_callback =
       base::BindOnce(&JsonRpcService::OnEthGetLogs,
                      weak_ptr_factory_.GetWeakPtr(), std::move(callback));
   RequestInternal(
       eth::eth_getLogs(from_block, to_block, std::move(contract_addresses),
-                       std::move(topics), ""),
+                       std::move(topics), block_hash),
       true, network_url, std::move(internal_callback));
```

Comment:
We don't need to look into params. Just change eth::eth_getLogs to something like:
```
std::string eth_getLogs(base::Value::Dict filter_options) {
  base::Value::List params;
  params.Append(std::move(filter_options));
  base::Value::Dict dictionary =
      GetJsonRpcDictionary("eth_getLogs", std::move(params));
  return GetJSON(dictionary);
}
```
If at some point we need a c++ way to create such filter then that should be 
```
base::Value::Dict MakeEthLogFilter(const std::string& from_block_quantity_tag,
                        const std::string& to_block_quantity_tag,
                        base::Value::List addresses,
                        base::Value::List topics,
                        const std::string& block_hash);
```
which makes a filter dict not a rpc call structure

Reply:
fixed